pkg/provider/github/repository.go

```diff
@@ -0,0 +1,110 @@
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func ConfigureRepository(ctx context.Context, run *params.Run, req *http.Request, payload string, logger *zap.SugaredLogger) (bool, bool, error) {
+	// gitea set x-github-event too, so skip it for the gitea driver
+	if h := req.Header.Get("X-Gitea-Event-Type"); h != "" {
+		return false, false, nil
+	}
+	event := req.Header.Get("X-Github-Event")
+	if event != "repository" {
+		return false, false, nil
+	}
+
+	eventInt, err := github.ParseWebHook(event, []byte(payload))
+	if err != nil {
+		return true, false, err
+	}
+	_ = json.Unmarshal([]byte(payload), &eventInt)
+	repoEvent, _ := eventInt.(*github.RepositoryEvent)
+
+	if repoEvent.GetAction() != "created" {
+		logger.Infof("github: repository event \"%v\" is not supported", repoEvent.GetAction())
+		return true, false, nil
+	}
+
+	logger.Infof("github: configuring repository cr for repo: %v", repoEvent.Repo.GetHTMLURL())
+	if err := createRepository(ctx, run.Info.Pac.AutoConfigureRepoNamespaceTemplate, run.Clients, repoEvent, logger); err != nil {
+		logger.Errorf("failed repository creation: %v", err)
+		return true, true, err
+	}
+
+	return true, true, nil
+}
+
+func createRepository(ctx context.Context, nsTemplate string, clients clients.Clients, gitEvent *github.RepositoryEvent, logger *zap.SugaredLogger) error {
+	repoNsName, err := generateNamespaceName(nsTemplate, gitEvent)
+	if err != nil {
+		return fmt.Errorf("failed to generate namespace for repo: %w", err)
+	}
+
+	logger.Info("github: generated namespace name: ", repoNsName)
+
+	// create namespace
+	repoNs := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: repoNsName,
+		},
+	}
+	repoNs, err = clients.Kube.CoreV1().Namespaces().Create(ctx, repoNs, metav1.CreateOptions{})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return fmt.Errorf("failed to create namespace %v: %w", repoNs.Name, err)
+	}
+
+	if errors.IsAlreadyExists(err) {
+		logger.Infof("github: namespace %v already exists, creating repository", repoNsName)
+	} else {
+		logger.Info("github: created repository namespace: ", repoNs.Name)
+	}
+
+	// create repository
+	repo := &v1alpha1.Repository{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      repoNsName,
+			Namespace: repoNsName,
+		},
+		Spec: v1alpha1.RepositorySpec{
+			URL: gitEvent.Repo.GetHTMLURL(),
+		},
+	}
+	repo, err = clients.PipelineAsCode.PipelinesascodeV1alpha1().Repositories(repoNsName).Create(ctx, repo, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create repository for repo: %v: %w", gitEvent.Repo.GetHTMLURL(), err)
+	}
+	logger.Infof("github: repository created: %s/%s ", repo.Namespace, repo.Name)
+	return nil
+}
+
+func generateNamespaceName(nsTemplate string, gitEvent *github.RepositoryEvent) (string, error) {
+	repoOwner, repoName, err := formatting.GetRepoOwnerSplitted(gitEvent.Repo.GetHTMLURL())
```

Comment:
GetRepoOwnerSplitter try to handle weird edge case for gitlab or other vcs naming, perhaps you can just use splitGithubURL which should be more robust for github

Reply:
GetRepoOwnerSplitter is doing a lot more things .. and it error out on normal git url like `https://github.com/openshift-pipelines/pipelines-as-code` GetRepoOwnerSplitted just split and return owner and repo which is fine I think here.

Reply:
errors out here https://github.com/openshift-pipelines/pipelines-as-code/blob/c2909334650ea316354d36c995252147bc7aab42/pkg/provider/github/github.go#L44
not sure why that condition is `<=` 🤔 any reason to have `=`

Reply:
it's at least org/repo/path, == would be no subpath too